build-tools/xaprepare/xaprepare/Application/Context.cs

```diff
@@ -329,6 +329,15 @@ partial class Context : AppObject
 		/// </summary>
 		public RefreshableComponent ComponentsToRefresh { get; set; }
 
+
+		public bool BuildingAZPPullRequest {
+			get {
+				string buildReason = Environment.GetEnvironmentVariable ("BUILD_REASON");
```

Comment:
Is there any way we can ensure that we are actually running on Azure here? The "danger" of this property being `true` on, say, developer's machine is low, but I'd still prefer to be thorough in the checks, if possible.

Reply:
I think the best way to do this would be to use environment variables, we could add an extra safety net here, but with the rework I don't think it's needed.